data/users/__init__.py

```diff
@@ -67,6 +67,7 @@ def get_users_handler(config, _, override_config_dir):
         ldap_user_filter = config.get("LDAP_USER_FILTER", None)
         ldap_superuser_filter = config.get("LDAP_SUPERUSER_FILTER", None)
         ldap_restricted_user_filter = config.get("LDAP_RESTRICTED_USER_FILTER", None)
+        ldap_referrals = int(config.get("LDAP_FOLLOW_REFERRALS", True))
```

Comment:
Should this be set by default. In my opinion, this should be set to False, and if required, explicitly set.

Reply:
@kleesc Since we follow ldap referrals today, albeit erroneously in most cases, wouldn't it make sense to preserve existing behavior and keep this set to `True` so we don't unintentionally break people on upgrade? I think the alternative is to risk breaking someone's existing config and forcing a callout/warning in the release notes. 

If we're confident that ldap referrals are genuinely broken, maybe it does make sense to change this to be default `False`?

Reply:
for the renuinely broken referrals ... we have one use case where referrals are working:

if for example the People Base DN `ou=people,dc=example,dc=com` is the referral itself, than our code will receive an ReferralException and handle it accordingly.

If the referral object is inside the search tree like `ou=External,ou=People,dc=example,dc=com` than we'll fail even at login.

As discussed, its 99.999% broken and we could provide a note with the Patch, that if someone configured it's base DN as referral, they should configure the referred LDAP uri instead as Base DN.